mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -249,6 +250,27 @@ object ParamValidators {
   def arrayLengthGt[T](lowerBound: Double): Array[T] => Boolean = { (value: Array[T]) =>
     value.length > lowerBound
   }
+
+  /**
+   * Checks that either inputCols and outputCols are set or inputCol and outputCol are set. If
+   * this is not true, an `IllegalArgumentException` is raised.
+   * @param model
+   */
+  private[spark] def checkMultiColumnParams(model: Params): Unit = {
+    model match {
+      case m: HasInputCols with HasInputCol if m.isSet(m.inputCols) && m.isSet(m.inputCol) =>
+        raiseIncompatibleParamsException("inputCols", "inputCol")
+      case m: HasOutputCols with HasOutputCol if m.isSet(m.outputCols) && m.isSet(m.outputCol) =>
+        raiseIncompatibleParamsException("outputCols", "outputCol")
+      case _ =>
+    }
```

Comment:
If we need to check other exclusive params, e.g., `inputCol` and `splitsArray` or `inputCols` and `splits`, why not just have a method like:

```scala
def checkExclusiveParams(model: Params, params: String*): Unit = {
  if (params.filter(model.isSet(_)).size > 1) {
    val paramString = params.mkString("`", "`, `", "`")
    throw new IllegalArgumentException(s"$paramString are exclusive, but more than one among them are set.")
  }
}
ParamValidators.checkExclusiveParams(this, "inputCol", "inputCols")
ParamValidators.checkExclusiveParams(this, "outputCol", "outputCols")
ParamValidators.checkExclusiveParams(this, "inputCol", "splitsArray")
ParamValidators.checkExclusiveParams(this, "inputCols", "splits")
```

Reply:
I added this method too in #20146.

Reply:
I think we can use that method once merged, thanks.

Reply:
I am not sure if #20146 will get merged for `2.3` - but I think we must merge this PR for `2.3` because I'd prefer not to have this inconsistency in param error handling between `QuantileDiscretizer` and `Bucketizer`. This is a relatively small change, so we can merge it into the branch if we move it quickly.

Reply:
Based on https://github.com/apache/spark/pull/20146#issuecomment-357068054 from @WeichenXu123, I think #20146 cannot get merged for 2.3.

Reply:
If this method looks good to you, maybe you can just copy it from #20146 to use here.

Reply:
@MLnick @viirya in order to address https://github.com/apache/spark/pull/19993/files#r161682506, I was thinking to let this method as it is (just renaming it as per @viirya suggestion) and only adding an `additionalExclusiveParams: (String, String)*` argument to the function. WDYT?

Reply:
I think @viirya's method is simpler and more general, so why not use it?

Reply:
@viirya your actual method in #20146 is slightly different (see [here](https://github.com/apache/spark/pull/20146/files#diff-72f95a0938e5a140d5126f06bdc381a6R254)). Is that the best version to use?

Reply:
@MLnick Yes. I didn't test the method posted here. The model possibly doesn't have the params, so we need to check it with `model.hasParam`. Please use the method in #20146.